ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/139__trusted_ca_bundles.robot

```diff
@@ -0,0 +1,155 @@
+
+    Log    message=Validate Trusted CA Bundle Management State Unmanaged    level=INFO
+
+    Set Trusted CA Bundle Management State    Unmanaged    ${RHOAI_OPERATOR_NS}
+
+    # Trusted CA BUndle managementStatus 'Unmanaged' should NOT result in bundle being overwirtten by operator
+    Set Custom CA Bundle Value On ConfigMap
+    ...    ${TRUSTED_CA_BUNDLE_CONFIGMAP}    random-ca-bundle-value    ${TEST_NS}
+    # Allow operator time to reconsile
+    Sleep    5
+    Wait Until Keyword Succeeds    3 min    0 sec
+    ...    Is CA Bundle Value Present    ${TRUSTED_CA_BUNDLE_CONFIGMAP}    random-ca-bundle-value    ${TEST_NS}    ${IS_PRESENT}
+
+    Log    message=Validate Trusted CA Bundle Management State Removed    level=INFO
+
+    Set Trusted CA Bundle Management State    Removed    ${RHOAI_OPERATOR_NS}
+
+    # Check that odh-trusted-ca-bundle has been 'Removed'
+    Wait Until Keyword Succeeds    3 min    0 sec
+    ...    Is Resource Present     ConfigMap    ${TRUSTED_CA_BUNDLE_CONFIGMAP}    ${TEST_NS}    ${IS_NOT_PRESENT}
+
+    Log    message=Restore DSCI to original state    level=INFO
+    Set Custom CA Bundle Value In DSCI   ''    ${RHOAI_OPERATOR_NS}
+    Set Trusted CA Bundle Management State    Managed    ${RHOAI_OPERATOR_NS}
+    Delete Namespace From Openshift    ${TEST_NS}
+
+
+*** Keywords ***
+Suite Setup
+    [Documentation]    Suite Setup
+    Log    message=Suite Setup.    level=INFO
+    RHOSi Setup
+
+Suite Teardown
+    [Documentation]    Suite Teardown
+    Log    message=Suite Teardown.    level=INFO
+    RHOSi Teardown
+
+Is Operator Available
+    [Documentation]    Checks if operator is available
+    [Arguments]    ${operator_name}    ${namespace}
+    ${rc}=    Run And Return Rc
+    ...    oc wait --timeout=2m --for condition=available -n ${namespace} deploy/${operator_name}
+    Should Be Equal As Integers    ${rc}     ${0}    msg=${operator_name} Not Available
+
+Is Resource Present
+    [Documentation]    Check If Resource Is Present In Namespace
+    [Arguments]       ${resource}     ${resource_name}    ${namespace}    ${expected_result}
+    ${rc}=     Run And Return Rc
+    ...  oc get ${resource} ${resource_name} -n ${namespace}
+    Should Be Equal    "${rc}"    "${expected_result}"    msg=${resource} does not exist in ${namespace}
+
+Is DSCI In Ready State
+    [Documentation]    Checks that DSCI Reconciled Succesfully
+    [Arguments]    ${dsci}    ${namespace}
+    ${rc}=    Run And Return Rc
+    ...    oc wait --timeout=3m --for jsonpath='{.status.conditions[].reason}'=ReconcileCompleted -n ${namespace} dsci ${dsci}
+    Should Be Equal As Integers    ${rc}     ${0}    msg=${dsci} not in Ready state
+
+Check ConfigMap Contains CA Bundle
+    [Documentation]    Checks that ConfigMap contains CA Bundle
+    [Arguments]    ${config_map}    ${ca_bundle_name}    ${namespace}
+    ${rc}=     Run And Return Rc
+    ...    oc get configmap ${config_map} -n ${namespace} -o yaml | grep ${ca_bundle_name}
```

Comment:
I believe this keyword should check that the value are actually present rather than just checking the key is there

Reply:
The keyword is doing as implemented, which is to check that the ConfigMap essentially has been created with expected 'key'. That said, revised keyword to be more specific. Also, will investigate how to validate the actual CA value for future PR.